### PR TITLE
[SW-1646] Run H2O Nodes With Security Parameters

### PR DESCRIPTION
**Pull request by mn-mikke**

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalH2OBackend.scala [hunk 1]
```diff
@@ -84,6 +84,30 @@ class ExternalH2OBackend(val hc: H2OContext) extends SparklingBackend with Exter
...
+    if (conf.jks.isDefined) {
+      cmdToLaunch ++ Seq("-jks", conf.jks.get)
+    }
+
+    if (conf.jksPass.isDefined) {
+      cmdToLaunch ++ Seq("-jks_pass", conf.jksPass.get)
+    }
+
+    if (conf.jksAlias.isDefined) {
+      cmdToLaunch ++ Seq("-jks_alias", conf.jksAlias.get)
```
Comment (mn-mikke) [inline core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalH2OBackend.scala]: https://github.com/h2oai/h2o-3/pull/3905

Comment (jakubhava) [reply core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalH2OBackend.scala]: Yup, good point, we need to expose it also on internal backend ( or just check if it's already exposed there)

File: core/src/main/scala/org/apache/spark/h2o/backends/internal/InternalBackendUtils.scala [hunk 2]
```diff
@@ -86,6 +86,7 @@ object InternalBackendUtils extends InternalBackendUtils {
 
     new ArgumentBuilder()
       .add(getH2OCommonArgs(conf))
+      .add(getLoginArgs(conf))
```
Comment (jakubhava) [inline core/src/main/scala/org/apache/spark/h2o/backends/internal/InternalBackendUtils.scala]: I know it's WIP -> just suggestion -> We could move `.add(getLoginArgs(conf))` to the `getH2OCommonArgs(conf)` as those are options shared with worker nodes & client and remove it from ` getH2OWorkerArgs` and `getH2OWorkerAsClientArgs`

File: core/src/main/scala/org/apache/spark/h2o/backends/SharedBackendUtils.scala [hunk 3]
```diff
@@ -154,6 +141,11 @@ private[backends] trait SharedBackendUtils extends Logging with Serializable {
       .add("-jks", conf.jks)
       .add("-jks_pass", conf.jksPass)
       .add("-jks_alias", conf.jksAlias)
+      .addIf("-hash_login", conf.hashLogin)
```
Comment (jakubhava) [inline core/src/main/scala/org/apache/spark/h2o/backends/SharedBackendUtils.scala]: I actually meant moving all the properties from `getLoginArgs` inside the method `getH2OCommonArgs` to ensure it's on both client & worker nodes

Comment (mn-mikke) [reply core/src/main/scala/org/apache/spark/h2o/backends/SharedBackendUtils.scala]: ah, wrong method. I originally wanted to move it to `getH2OCommonArgs`. My bad, thanks!

Comment (jakubhava) [reply core/src/main/scala/org/apache/spark/h2o/backends/SharedBackendUtils.scala]: 👍

File: core/src/main/scala/org/apache/spark/h2o/backends/SharedBackendUtils.scala [hunk 4]
```diff
@@ -123,12 +123,9 @@ private[backends] trait SharedBackendUtils extends Logging with Serializable {
       .add("-nthreads", Some(conf.nthreads).filter(_ > 0).orElse(conf.sparkConf.getOption("spark.executor.cores")))
       .add("-internal_security_conf", conf.sslConf)
       .add("-client_disconnect_timeout", conf.clientCheckRetryTimeout)
-      .buildArgs()
-  }
-
-
-  def getLoginArgs(conf: H2OConf): Seq[String] = {
-    new ArgumentBuilder()
+      .add("-jks", conf.jks)
```
Comment (mn-mikke) [inline core/src/main/scala/org/apache/spark/h2o/backends/SharedBackendUtils.scala]: Still need to look after the distribution of jks to each node.

**Comment by mn-mikke** [comment]
@jakubhava I also updated the documentation, please can you take a look one more time.

File: doc/src/site/sphinx/tutorials/kerberos_auth.rst [hunk 5]
```diff
@@ -21,10 +21,8 @@ Generally, to enable Kerberos authentication you need to set the following envir
  - ``spark.ext.h2o.login.conf=kerberos.conf``
  - ``spark.ext.h2o.user.name=username``
 
-where ``kerberos.conf`` is the configuration file for the Kerberos connection.
-
-The last option is only needed in cases when your user name differs from your Kerberos user name. The user name in
-Sparkling Water needs to be the same as is in the Kerberos to enable the authentication.
+where ``kerberos.conf`` is the configuration file for the Kerberos connection and zusername` is a username of your Kerberos account
```
Comment (jakubhava) [inline doc/src/site/sphinx/tutorials/kerberos_auth.rst]: typo: `zusername`

**Comment by jakubhava** [comment]
Thank you 🍺 !

**Pull request merged**